scripts/filter_expected_warnings.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+"""Script to filter out old warnings from doxygen."""
+
+import sys
+import re
+
+
+def compare_lists(list1, list2):
+    """Simple comparison of two lists."""
+    if len(list1) != len(list2):
+        return False
+    for (elt1, elt2) in zip(list1, list2):
+        if elt1 != elt2:
+            return False
+    return True
+
+
+class DoxygenWarning(object):
+    """Doxygen warning class."""
+
+    def __init__(self, firstline, filename, warning):
+        self.firstline = firstline
+        self.filename = filename
+        self.warning = warning
+        self.otherlines = []
+
+    def approx_equals(self, other):
```

Comment:
Why is this called `approx_equals`? It seems to do exact equality?

(It's been pointed out to me that this apparently ignores the _path_ to the file and the exact line numbers I think?)

Reply:
That's correct - it doesn't compare everything in the first line - ignoring the path and line numbers.  The path will depend on where Doxygen is run, and the line numbers will change when lines are added or removed elsewhere in the source files.

Reply:
I suppose it could have a more descriptive name, like `equals_ignoring_path_and_line_number`